File: migrate.py
import os
import re

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    if 'getDbData' not in content and 'saveDbData' not in content:
        return

    new_content = content

    new_content = re.sub(r'(?<!await\s)getDbData\(\)', 'await getDbData()', new_content)
    new_content = re.sub(r'(?<!await\s)saveDbData\(', 'await saveDbData(', new_content)

    lines = new_content.split('\n')
    for i, line in enumerate(lines):
        if 'function' in line and not 'async ' in line and ('Page' in line or 'GET' in line or 'PUT' in line or 'POST' in line or 'Client' not in line):
            if 'export default function' in line or 'export function' in line:
                lines[i] = line.replace('function', 'async function')
        # Also fix any "export async function saveX(formData: FormData)" where getDbData inside wasn't awaited, but they already had async.

    new_content = '\n'.join(lines)

    if new_content != content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Updated {filepath}")

def main():
    for root, dirs, files in os.walk('src'):
        for file in files:
            if file.endswith(('.ts', '.tsx')) and 'db.ts' not in file:
                process_file(os.path.join(root, file))

if __name__ == '__main__':
    main()

File: migrate.js
const fs = require('fs');
const path = require('path');

function processFile(filepath) {
    let content = fs.readFileSync(filepath, 'utf-8');

    if (!content.includes('getDbData') && !content.includes('saveDbData')) {
        return;
    }

    let newContent = content;

    // Replace getDbData() with await getDbData() if not preceded by await
    newContent = newContent.replace(/(?<!await\s)getDbData\(\)/g, 'await getDbData()');
    
    // Replace saveDbData( with await saveDbData( if not preceded by await
    newContent = newContent.replace(/(?<!await\s)saveDbData\(/g, 'await saveDbData(');

    let lines = newContent.split('\n');
    for (let i = 0; i < lines.length; i++) {
        let line = lines[i];
        if (line.includes('function') && !line.includes('async ') && (line.includes('Page') || line.includes('GET') || line.includes('PUT') || line.includes('POST') || !line.includes('Client'))) {
            if (line.includes('export default function') || line.includes('export function')) {
                lines[i] = line.replace('function', 'async function');
            }
        }
    }

    newContent = lines.join('\n');

    if (newContent !== content) {
        fs.writeFileSync(filepath, newContent, 'utf-8');
        console.log(`Updated ${filepath}`);
    }
}

function walkArgs(dir) {
    fs.readdirSync(dir).forEach(file => {
        let fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walkArgs(fullPath);
        } else if (fullPath.endsWith('.ts') || fullPath.endsWith('.tsx')) {
            if (!fullPath.includes('db.ts')) {
                processFile(fullPath);
            }
        }
    });
}

walkArgs('src');
